subworkflows/local/utils_nfcore_methylseq_pipeline/main.nf

```diff
@@ -0,0 +1,268 @@
+*/
+
+workflow PIPELINE_INITIALISATION {
+
+    take:
+    version           // boolean: Display version and exit
+    help              // boolean: Display help text
+    validate_params   // boolean: Boolean whether to validate parameters against the schema at runtime
+    monochrome_logs   // boolean: Do not use coloured log outputs
+    nextflow_cli_args //   array: List of positional nextflow CLI args
+    outdir            //  string: The output directory where the results will be saved
+    input             //  string: Path to input samplesheet
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    //
+    // Print version and exit if required and dump pipeline parameters to JSON file
+    //
+    UTILS_NEXTFLOW_PIPELINE (
+        version,
+        true,
+        outdir,
+        workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1
+    )
+
+    //
+    // Validate parameters and generate parameter summary to stdout
+    //
+    pre_help_text = nfCoreLogo(monochrome_logs)
+    post_help_text = '\n' + workflowCitation() + '\n' + dashedLine(monochrome_logs)
+    def String workflow_command = "nextflow run ${workflow.manifest.name} -profile <docker/singularity/.../institute> --input samplesheet.csv --outdir <OUTDIR>"
+    UTILS_NFVALIDATION_PLUGIN (
+        help,
+        workflow_command,
+        pre_help_text,
+        post_help_text,
+        validate_params,
+        "nextflow_schema.json"
+    )
+
+    //
+    // Check config provided to the pipeline
+    //
+    UTILS_NFCORE_PIPELINE (
+        nextflow_cli_args
+    )
+    //
+    // Custom validation for pipeline parameters
+    //
+    validateInputParameters()
+
+    //
+    // Create channel from input file provided through params.input
+    //
+    Channel
+        .fromSamplesheet("input")
+        .map {
+            meta, fastq_1, fastq_2 ->
+            if (!fastq_2) {
+                return [ meta + [ single_end:true ], [ fastq_1 ] ]
+            } else {
+                return [ meta + [ single_end:false ], [ fastq_1, fastq_2 ] ]
+            }
+        }
+        .groupTuple()
+        .map {
+            meta, fastq ->
+            def meta_clone = meta.clone()
+            parts = meta_clone.id.split('_')
+            meta_clone.id = parts.length > 1 ? parts.join('_') : meta_clone.id
+            [ meta_clone, fastq ]
+        }
+        .groupTuple(by: [0])
+        .branch {
+            meta, fastq ->
+            single: fastq.size() == 1
+            return [ meta, fastq.flatten() ]
+            multiple: fastq.size() > 1
+            return [ meta, fastq.flatten() ]
+        }
+        .set { ch_fastq }
+
+    //
+    // MODULE: Concatenate FastQ files from same sample if required
+    //
+    CAT_FASTQ ( ch_fastq.multiple )
+        .reads
+        .mix(ch_fastq.single)
+        .set { ch_samplesheet }
+    ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first())
```

Comment:
Is this combining multiple runs for the same sample? If so, this seems like a much more concise way than how I am doing it currently!

Reply:
Exactly! Borrowed from rnaseq. We should probably make it a subworkflow at this point 🤔